flax/optim/radam.py

```diff
@@ -0,0 +1,111 @@
+# Lint as: python3
+
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .. import struct
+
+import jax.numpy as jnp
+from jax import lax
+
+import numpy as onp
+
+from .base import OptimizerDef
+
+
+@struct.dataclass
+class _RAdamHyperParams:
+  learning_rate: onp.ndarray
+  beta1: onp.ndarray
+  beta2: onp.ndarray
+  eps: onp.ndarray
+  weight_decay: onp.ndarray
+
+
+@struct.dataclass
+class _RAdamParamState:
+  grad_ema: onp.ndarray
+  grad_sq_ema: onp.ndarray
+
+
+class RAdam(OptimizerDef):
+  """RAdam optimizer."""
```

Comment:
In particular for more niche optimizers I thing it is good to have a better description of the algorithm.
Explaining the algorithm and a link to the paper

Reply:
I'll definitely add more clear documentation. Thanks a lot for the reviews. I apologize for the really late reply.